refactor: changed survey report message location and added a new info…

Asespinel opened the pull request:
**Description**

This pull request introduces a refactor of the Survey Report management in the Django admin site. The changes include two key modifications aimed at enhancing the clarity and functionality of the admin interface.

1. Moving Survey Report Banner to the "Messages" Section

- The Survey Report Banner, previously located in the header section, has now been moved to the "Messages" side of the admin site.
- This change aims to make the banner  more intuitive and user-friendly.

2. Adding a New State "Sent" to Reports

- A new state, "Sent," has been added to the reports.
- This state is applied to reports that have been successfully sent to Axim.
- Previously, only the state "generated" was implemented which could be confusing and didn't confirm if the report was sent or not.
- The main difference with 
https://github.com/openedx/edx-platform/pull/34024
is that we don't touch the models file and we just change the state column to show the "sent" status based on the SurveyReportUpload status code.

Testing

1. You can click on the "send report" button to send the report and a confirmation message should appear
2. You can also click on the "Dismiss" button to close the banner message
3. You should see the recently sent message under {{domain}}/admin/survey_report/ with the state "sent"

Example:
![survey_report](https://github.com/openedx/edx-platform/assets/79876430/0d97ce34-3204-428f-9c7d-ff8ec6523b28)

Asespinel committed:
refactor: changed survey report message location and added a new informative sent state

openedx-webhooks commented:
Thanks for the pull request, @Asespinel! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

Asespinel committed:
chore: added Generate and Send text on button

felipemontoya commented on openedx/features/survey_report/templates/survey_report/admin_banner.html:
```diff
@@ -31,6 +31,7 @@ <h1 style="margin: 0; color: #FFFF; font-weight: 600;">Join the Open edX Data Sh
 {% endif %}
 
     <!-- The original content of the block -->
+    <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>
```
Should we load jquery using the file in the vendor dir?
Such as:
https://github.com/openedx/edx-platform/blob/ff3d21f186a8acfcc441e74a5edcf661ea89b873/lms/templates/verify_student/test/fake_softwaresecure_response.html#L24

Asespinel replied:
Sure thing I changed the jquery accordingly.

felipemontoya commented on openedx/features/survey_report/admin.py:
```diff
@@ -80,4 +80,10 @@ def get_actions(self, request):
         return actions
 
+    def report_state(self, obj):
+        if obj.surveyreportupload_set.last().is_uploaded():
```
since we are doing this to avoid changing the model at this stage, can we at least leave a comment explaining that?

Asespinel replied:
Added comment on the method.

Asespinel committed:
chore: added comment to new state column and added reference to static jquery.

felipemontoya reviewed (approved):
Thanks, lgtm. The wording is clearer now and the banner is in a better location

MaferMazu commented:
Thanks for this refactor @Asespinel. The changes you proposed make sense. I have two comments:

- Can you add an end new line in the `admin/base_site.html` file?
- Can we add some conditional before `if obj.surveyreportupload_set.last().is_uploaded():` in the `report_state` function? To avoid `AttributeError: 'NoneType' object has no attribute 'is_uploaded'`. I think I had that error because I don't have a survey report yet.

Asespinel committed:
fix: fixed non Attribute error when a report hasn't been sent yet

Asespinel commented:
Thanks for the help @MaferMazu I implemented the fix for your problem and I also added the blank line as requested. Let me know if you need anything else.

MaferMazu reviewed (approved):
This works as expected. Thanks for the fix @Asespinel.

Can we merge this, or should we wait for comments by @ormsbee?

ormsbee commented:
No need to wait on my review. Thank you.

openedx-webhooks commented:
@Asespinel 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/34033&entry.1484655318=https://github.com/Asespinel&entry.752974735=2024-01-09+22:54&entry.1917517419=2024-01-16+21:25&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
